### PR TITLE
Families wanted stars now only show up when you are involved in it, and announcements for wanted level changes

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus
 add: Families wanted stars now only show up when you are involved in it, and announcements for wanted level changes
 add: hovering over wanted stars tells you what they are through a tooltip
/:cl:

also cleaned up the process() code a bit, more event based now.

### CODE REVIEW

File: code/game/gamemodes/gang/gang.dm [hunk 1]
```diff
@@ -238,61 +200,135 @@ GLOBAL_VAR_INIT(deaths_during_shift, 0)
...
+	var/announcement_message
+	switch(newlevel)
+		if(1)
+			announcement_message = "There are now only a few police vehicle headed towards [station_name()]."
+		if(2)
+			announcement_message = "There seem to be fewer police vehicles headed towards [station_name()]."
+		if(3)
+			announcement_message = "There are no longer top-trained agents in the fleet headed towards [station_name()]."
+		if(4)
+			announcement_message = "The convoy enroute to [station_name()] seems to no longer consists of national guard personnel."
```
Comment (trollbreeder) [inline code/game/gamemodes/gang/gang.dm]: oops, typo
```suggestion
			announcement_message = "The convoy enroute to [station_name()] seems to no longer consist of national guard personnel."
```

File: code/game/gamemodes/gang/gang.dm [hunk 2]
```diff
@@ -238,61 +200,135 @@ GLOBAL_VAR_INIT(deaths_during_shift, 0)
...
+///Updates the icon states for everyone and sends outs announcements regarding the police.
+/datum/game_mode/gang/proc/update_wanted_level(newlevel)
+	if(newlevel > wanted_level)
+		on_gain_wanted_level(newlevel)
+
+	else if (newlevel < wanted_level)
+		on_lower_wanted_level(newlevel)
+	for(var/mob/M in GLOB.player_list)
+		if(!M.hud_used)
+			continue
```
Comment (Rohesie) [inline code/game/gamemodes/gang/gang.dm:245]:
```suggestion
	for(var/i in GLOB.player_list)
		var/mob/M = i
		if(!M.hud_used)
			continue
```

File: code/_onclick/hud/families.dm [hunk 3]
```diff
@@ -0,0 +1,26 @@
+/obj/screen/wanted
+	name = "Space Police Alertness"
+	desc = "Shows the current level of hostility the space police is planning to rain down on you. Better be careful."
+	icon = 'icons/obj/gang/wanted_160x32.dmi'
+	icon_state = "wanted_0"
+	screen_loc = ui_wanted_lvl
+	var/level
+	var/cops_arrived
```
Comment (Rohesie) [inline code/_onclick/hud/families.dm:8]:
```suggestion
	///Wanted level, affects the hud icon.
	var/level
	///Boolean, have the cops arrived? If so, the icon stops changing and remains the same.
	var/cops_arrived
```

Comment (Qustinnus) [reply code/_onclick/hud/families.dm:8]: thansk forgot to do it myself :)

File: code/game/gamemodes/gang/gang.dm [hunk 4]
```diff
@@ -238,61 +200,136 @@ GLOBAL_VAR_INIT(deaths_during_shift, 0)
...
+///Updates the icon states for everyone and sends outs announcements regarding the police.
+/datum/game_mode/gang/proc/update_wanted_level(newlevel)
+	if(newlevel > wanted_level)
+		on_gain_wanted_level(newlevel)
+
+	else if (newlevel < wanted_level)
+		on_lower_wanted_level(newlevel)
+	for(var/i in GLOB.player_list)
+		var/mob/M = i
+		if(!M.hud_used)
```
Comment (Rohesie) [inline code/game/gamemodes/gang/gang.dm:245]:
```suggestion
		if(!M.hud_used?.wanted_lvl)
```

File: code/modules/antagonists/gang/gang.dm [hunk 5]
```diff
@@ -18,12 +18,20 @@
...
 
 
 /datum/antagonist/gang/remove_innate_effects(mob/living/mob_override)
 	package_spawner.Remove(owner.current)
 	var/mob/living/M = mob_override || owner.current
 	remove_antag_hud(antag_hud_type, M)
+	if(M.hud_used)
+		var/datum/hud/H = M.hud_used
+		H.infodisplay -= H.wanted_lvl
+		qdel(H.wanted_lvl)
```
Comment (Rohesie) [inline code/modules/antagonists/gang/gang.dm:34]:
```suggestion
		QDEL_NULL(H.wanted_lvl)
```

File: code/modules/antagonists/ert/ert.dm [hunk 6]
```diff
@@ -205,11 +205,19 @@
...
+		H.infodisplay += H.wanted_lvl
 
 
 /datum/antagonist/ert/families/remove_innate_effects(mob/living/mob_override)
 	var/mob/living/M = mob_override || owner.current
 	remove_antag_hud(antag_hud_type, M)
+	if(M.hud_used)
+		var/datum/hud/H = M.hud_used
+		H.infodisplay -= H.wanted_lvl
+		qdel(H.wanted_lvl)
```
Comment (Rohesie) [inline code/modules/antagonists/ert/ert.dm:220]:
```suggestion
		QDEL_NULL(H.wanted_lvl)
```

**Pull request merged**